tests/scripts/visual-coverage.sh

```diff
@@ -1,16 +1,24 @@
 #!/bin/bash
 
-# start Storybook in the same shell
-. ./tests/scripts/start-storybook.sh
+#check for existing process on port:6006
+process=`lsof -i :6006`
```

Comment:
Instead of guarding this script by whether 6006 is available, can we tweak start-storybook.sh to be a no-op if 6006 is not available? That way if storybook is already running we don't kill it and restart unnecessarily

Reply:
Is there a case where a process is running on port 6006 but it's not storybook? If there is, I am concerned about the `test-storybook --coverage` step in visual test script erroring out.

Reply:
Also to confirm, if there is already a storybook instance running on port 6006, we'd start a visual test using that process. Do we need to kill it after we run the test or leave it as it since it's running before we've started the test?

Reply:
I wouldn't worry about a case where something is running on 6006 but it's not storybook. Since this item was for local dev QOL imo it's not a big deal if we false start.

I think ideally, if storybook was started before the test we would leave it, and if storybook was started solely for the purposes of visual-coverage.sh we would kill it. I would do something like "if start-storybook is a no-op return a blank JOB_ID, and if JOB_ID is blank don't kill it". I'm not exactly sure how returning stuff works for bash scripts.

Reply:
> I'm not exactly sure how returning stuff works for bash scripts.

because we run the start-storybook script in the same shell the variables persist between the scripts. so defining JOB_ID in one will let us use/modify it in the other

Reply:
> I wouldn't worry about a case where something is running on 6006 but it's not storybook. Since this item was for local dev QOL imo it's not a big deal if we false start.
> 
> I think ideally, if storybook was started before the test we would leave it, and if storybook was started solely for the purposes of visual-coverage.sh we would kill it. I would do something like "if start-storybook is a no-op return a blank JOB_ID, and if JOB_ID is blank don't kill it". I'm not exactly sure how returning stuff works for bash scripts.

I modified the behavior to kill storybook only when it's started by the test. So the current behavior is to assume existing process on port is storybook and start test using the storybook instance, then leave the storybook running if it's a preexisting process.

Reply:
It turns out that the storybook processes running are the child processes for the original process with the JOB_ID, so we have to kill them along with the parent process.